lexer/lexer.go

```diff
@@ -150,9 +152,10 @@ func (l *Lexer) NextToken() token.Token {
 			tok = l.newToken(token.SLASH)
 		}
 	case '#':
-		tok.Type = token.COMMENT
-		tok.Position = l.position
-		tok.Literal = l.readLine()
+		// comment chunk - skip it
+		_ = l.readLine()
+		l.readChar()
+		goto getNextToken
```

Comment:
I'm not a big fan of GOTOs -- I think we should be able to just call ourselves to go to the next token?

Reply:
Sure. Just trying to avoid any issues with the lack of tail-call elimination in Go.